File: codingjs/linearIn.js
function linearIn(outer, inner){
  let prevIndex = 0;
  for (let i = 0; i < inner.length; i++) {
    if (outer.indexOf(inner[i]) >= prevIndex) {
      prevIndex = outer.slice(prevIndex).indexOf(inner[i]);
    } else {
      return false;
    }
  }
  return true;
}

File: codingbat/java/linearIn.java
public boolean linearIn(int[] outer, int[] inner) {
  // outerとinnerを同時に見て、
  // innerのほうが大きかったらouterのindexを大きくして、
  // outerのほうが大きかったらfalse
  // それ以外（inner == outer）のときはinnernのindexを大きくする
  // （outerも大きくしてしまうとinner = [2, 2], outer = [2, 3]のような場合にfalseになってしまう）
  int innerIndex = 0;
  int outerIndex = 0;
  while (innerIndex < inner.length && outerIndex < outer.length) {
    if (inner[innerIndex] > outer[outerIndex]) {
      outerIndex++;
    } else if (inner[innerIndex] < outer[outerIndex]) {
      return false;
    } else {
      innerIndex++;
    }
  }
  return innerIndex >= inner.length;
}:
